### PR TITLE
[+] Environment - Add feedback that the delete process has succeeded or failed

### PR DESCRIPTION
**Pull request by root-io**

### CODE REVIEW

**Comment by root-io** [comment]
[(2) Environment Destroy - Users using the CLI want to have the feedback that the process has succeeded or failed.](https://trello.com/c/gTjwviNQ/3094-2-environment-destroy-users-using-the-cli-want-to-have-the-feedback-that-the-process-has-succeeded-or-failed)

File: src/commands/environments/delete.js [hunk 1]
```diff
@@ -30,8 +30,12 @@ class DeleteCommand extends Command {
       }
 
       if (!answers || answers.confirm === environment.name) {
-        await manager.deleteEnvironment(config.environmentId);
-        console.log(`Environment ${chalk.red(environment.name)} successfully deleted.`);
+        const deleteEnvironment = await manager.deleteEnvironment(config.environmentId);
+        if (deleteEnvironment) {
+          console.log(`Environment ${chalk.red(environment.name)} successfully deleted.`);
```
Comment (VincentMolinie) [inline src/commands/environments/delete.js]:
```suggestion
          this.log(`Environment ${chalk.red(environment.name)} successfully deleted.`);
```

File: src/commands/environments/delete.js [hunk 2]
```diff
@@ -30,8 +30,12 @@ class DeleteCommand extends Command {
...
+        } else {
+          logger.error('💀  Oops, something went wrong.💀');
```
Comment (VincentMolinie) [inline src/commands/environments/delete.js]:
```suggestion
          this.error('💀  Oops, something went wrong.💀', { exit: 1 });
```

File: src/services/environment-manager.js [hunk 3]
```diff
@@ -10,6 +10,29 @@ const { promisify } = require('util');
...
+      .set('forest-origin', 'Lumber')
+      .set('forest-project-id', config.projectId);
+
+    if (jobResponse
+        && jobResponse.body
+        && jobResponse.body.state
+        && jobResponse.body.state !== 'inactive'
+        && jobResponse.body.state !== 'active') {
+      cli.action.stop();
+      if (jobResponse.body.state === 'failed') {
```
Comment (VincentMolinie) [inline src/services/environment-manager.js]: can be simplified

File: src/services/environment-manager.js [hunk 4]
```diff
@@ -55,12 +78,17 @@ function EnvironmentManager(config) {
...
-    return agent
+    const deleteEnvironmentResponse = await agent
       .del(`${config.serverHost}/api/environments/${environmentId}`)
       .set('Authorization', `Bearer ${authToken}`)
       .set('forest-origin', 'Lumber')
-      .set('forest-project-id', config.projectId)
-      .then(() => true);
+      .set('forest-project-id', config.projectId);
+
+    const jobId = deleteEnvironmentResponse.body.meta.job_id;
```
Comment (VincentMolinie) [inline src/services/environment-manager.js]: Check for errors

File: test/commands/environments/delete.test.js [hunk 5]
```diff
@@ -6,37 +6,51 @@ const fancy = test.register('nock', Nock);
...
+          apiEndpoint: 'https://forestadmin-server-staging.herokuapp.com',
+          active: true,
+          type: 'development',
+          liana: 'forest-express-sequelize',
+          version: '1.3.2',
+          FOREST_ENV_SECRET: '2c38a1c6bb28e7bea1c943fac1c1c95db5dc1b7bc73bd649a0b113713ee29125',
+        })))
+      .nock('http://localhost:3001', api => api
+        .delete('/api/environments/324')
+        .reply(204, {
```
Comment (VincentMolinie) [inline test/commands/environments/delete.test.js]: 204 with meta ?

File: test/commands/environments/delete.test.js [hunk 6]
```diff
@@ -6,37 +6,51 @@ const fancy = test.register('nock', Nock);
...
+            job_id: 78,
+          },
+        }))
+      .nock('http://localhost:3001', api => api
+        .get('/api/jobs/78')
+        .reply(200, {
+          state: 'complete',
+          progress: '100',
+        }))
+      .command(['environments:delete', '324', '-p', '82', '--force'])
```
Comment (VincentMolinie) [inline test/commands/environments/delete.test.js]: test a failed job

File: test/commands/environments/delete.test.js [hunk 7]
```diff
@@ -6,37 +6,51 @@ const fancy = test.register('nock', Nock);
...
+      .it('should delete the environment', (ctx) => {
+        expect(ctx.stdout).to.contain('Environment Staging successfully deleted.');
+      });
+  });
```
Comment (VincentMolinie) [inline test/commands/environments/delete.test.js]: Could we have a test with an active state and then se the server to reply complete or failed to see if it works correctly ?

**Pull request merged**